library/std/src/sys/unix/thread.rs

```diff
@@ -345,6 +345,33 @@ pub fn available_parallelism() -> io::Result<NonZeroUsize> {
                 }
             }
 
+            #[cfg(target_os = "netbsd")]
+            {
+                unsafe {
+                    let set = libc::_cpuset_create();
+                    if !set.is_null() {
+                        let mut count: usize = 0;
+                        if libc::pthread_getaffinity_np(libc::pthread_self(), libc::_cpuset_size(set), set) == 0 {
+                            #[cfg(target_pointer_width = "32")]
+                            const MAXCPUS: u64 = 32;
+                            #[cfg(target_pointer_width = "64")]
+                            const MAXCPUS: u64 = 256;
+                            for i in 0..MAXCPUS {
```

Comment:
I'm wary of hard-coding `MAXCPUS`. Would it be sufficient to just count unbounded (or to `u64::MAX`) and let `-1 => break` do its job?